implementations/micrometer-registry-azure/src/main/java/io/micrometer/azure/AzureMeterRegistry.java

```diff
@@ -0,0 +1,301 @@
+        this(config, clock, configuration, Executors.defaultThreadFactory());
+    }
+
+    private AzureMeterRegistry(AzureConfig config, Clock clock, @Nullable TelemetryConfiguration clientConfig, ThreadFactory threadFactory) {
+        super(config, clock);
+
+        config().namingConvention(new AzureNamingConvention());
+
+        if (clientConfig == null) {
+            // Get the current active instance of TelemetryConfiguration.
+            // This will only happen for Non SpringBoot scenario.
+            clientConfig = TelemetryConfiguration.getActive();
+        }
+
+        if (StringUtils.isEmpty(clientConfig.getInstrumentationKey())) {
+
+            // Pick Instrumentation Key From the Config if not set via instrumentationKey / Environment Variable
+            clientConfig.setInstrumentationKey(config.instrumentationKey());
+        }
+
+        requireNonNull(clientConfig);
+        requireNonNull(clientConfig.getInstrumentationKey());
+
+        this.client = new TelemetryClient(clientConfig);
+        start(threadFactory);
+    }
+
+    @Override
+    protected void publish() {
+            getMeters().forEach(meter -> {
+                try {
+                        Meter.Id id = meter.getId();
+                        String name = getConventionName(id);
+                        Map<String, String> properties = getConventionTags(id).stream()
+                            .collect(Collectors.toMap(Tag::getKey, Tag::getValue));
+
+                        if (meter instanceof TimeGauge) {
+                            trackGauge(name, properties, (TimeGauge) meter);
+                        } else if (meter instanceof Gauge) {
+                            trackGauge(name, properties, ((Gauge) meter));
+                        } else if (meter instanceof Counter) {
+                            trackCounter(name, properties, (Counter) meter);
+                        } else if (meter instanceof FunctionCounter) {
+                            trackCounter(name, properties, ((FunctionCounter) meter));
+                        } else if (meter instanceof Timer) {
+                            trackTimer(name, properties, ((Timer) meter));
+                        } else if (meter instanceof FunctionTimer) {
+                            trackTimer(name, properties, ((FunctionTimer) meter));
+                        } else if (meter instanceof DistributionSummary) {
+                            trackDistributionSummary(name, properties, ((DistributionSummary) meter));
+                        } else if (meter instanceof LongTaskTimer) {
+                            trackLongTaskTimer(id, properties, ((LongTaskTimer) meter));
+                        } else {
+                            trackMeter(name, properties, meter);
+                        }
+                } catch (Exception e) {
+                    logger.trace(String.format("Failed to track metric with name %s", getConventionName(meter.getId())));
```

Comment:
should this message be marked as warning or error?

Reply:
A warning makes sense. Error level could be for something fatal happening in the customer code itself. I will change this to warning level. Possibly add a trackTrace() call also for investigation.

Reply:
updated.